### PR TITLE
Mark functions and variables as static

### PR DESCRIPTION
**Pull request by MaddTheSane**
These are variables and functions that are only used in their own source file.

This also moves some `extern` declarations in source files to relevant headers.
I also marked functions that I couldn't find any place it was used in Descent 3 (mostly light map generation) but might be used elsewhere (The editor, perhaps?).

### CODE REVIEW

File: 2dlib/font.cpp [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-*
-* This program is distributed in the hope that it will be useful,
-* but WITHOUT ANY WARRANTY; without even the implied warranty of
-* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-* GNU General Public License for more details.
-*
-* You should have received a copy of the GNU General Public License
-* along with this program.  If not, see <http://www.gnu.org/licenses/>.
-*/
+ * Descent 3
```
Comment (Lgt2x) [inline 2dlib/font.cpp:2]: Please revert all license formatting changes

Comment (MaddTheSane) [reply 2dlib/font.cpp]: Done.

File: scripts/osiris_common.h [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
```
Comment (Jayman2000) [inline scripts/osiris_common.h:17]: It looks like one of the copyright headers still hasn’t been fixed.

**Review by Jayman2000 — CHANGES_REQUESTED** [review]
I can confirm that this PR builds and runs fine on NixOS 23.11.

**Comment by MaddTheSane** [comment]
I must have missed a few…

Sorry.

File: scripts/osiris_common.h (hunk 2 above)
Comment (MaddTheSane) [reply scripts/osiris_common.h:17]: I must have missed a few…

Sorry.

**Review by Lgt2x — COMMENTED** [review]
Good work. I went through the diff and it looks good.

Note to the merger to squash commits

**Review by Lgt2x — APPROVED** [review]

**Review by Jayman2000 — APPROVED** [review]

**Pull request merged**